rust-app/Cargo.toml

```diff
@@ -11,42 +11,39 @@ arrayvec = { version = "0.7.2", default-features = false }
 ledger-crypto-helpers = { git = "https://github.com/alamgu/ledger-crypto-helpers" }
 ledger-log = { git = "https://github.com/alamgu/ledger-log" }
 ledger-parser-combinators = { git = "https://github.com/alamgu/ledger-parser-combinators", branch="async-split-take-2" }
-alamgu-async-block = { git = "https://github.com/alamgu/alamgu-async-block", branch = "prompt-queueing" }
+alamgu-async-block = { git = "https://github.com/alamgu/alamgu-async-block" }
 zeroize = { version = "1.5.2", default-features = false }
 num_enum = { version = "0.5", default-features = false }
 pin-cell = "0.2.0"
 pin-project = "1.0.10"
-include_gif = { git = "https://github.com/LedgerHQ/sdk_include_gif" }
+include_gif = "1.0.1"
 
 # Pass --features speculos,ledger-log/log_info (or log_trace, etc) to enable
 # speculos logging and change log level.
 [features]
 default = [ ]
 # Pass --features speculos,ledger-log/log_info (or log_trace, etc) to enable speculos logging and change log level.
-speculos = [ "nanos_sdk/speculos", "ledger-log/speculos", "ledger-log/log_error", "ledger-parser-combinators/logging" ]
+speculos = [ "ledger_device_sdk/speculos", "ledger-log/speculos", "ledger-log/log_error", "ledger-parser-combinators/logging" ]
 extra_debug = ["ledger-log/log_trace"]
-pending_review_screen = ["nanos_sdk/pending_review_screen"]
 
 [target.'cfg(target_family = "bolos")'.dependencies]
-nanos_sdk = { git = "https://github.com/LedgerHQ/ledger-nanos-sdk.git" }
-nanos_ui = { git = "https://github.com/LedgerHQ/ledger-nanos-ui.git" }
+ledger_device_sdk = "1.5.1"
+ledger_secure_sdk_sys = "1.2.0"
 ledger-prompts-ui = { git = "https://github.com/alamgu/ledger-prompts-ui" }
 
-[target.'cfg(target_family = "bolos")'.dev-dependencies.nanos_sdk]
-git = "https://github.com/LedgerHQ/ledger-nanos-sdk.git"
+[target.'cfg(target_family = "bolos")'.dev-dependencies.ledger_device_sdk]
+version = "1.5.1"
 features = [ "speculos" ]
 
-[dev-dependencies]
-testmacro = { git = "https://github.com/yhql/testmacro" }
-
 [[bin]]
 name = "sui"
 test = false
 path = "bin-src/main.rs"
 
-[patch."https://github.com/LedgerHQ/ledger-nanos-sdk.git".nanos_sdk]
-git = "https://github.com/alamgu/ledger-nanos-sdk.git"
-branch = "relocating-loader-w-fixes"
+[patch.crates-io]
+ledger_device_sdk = { git = "https://github.com/alamgu/ledger-nanos-sdk.git", branch = "alamgu-branch" }
+ledger_secure_sdk_sys = { git = "https://github.com/alamgu/ledger-nanos-sdk.git", branch = "alamgu-branch" }
+include_gif = { git = "https://github.com/alamgu/ledger-nanos-sdk.git", branch = "alamgu-branch" }
```

Comment:
Any reason why this is still needed?

Reply:
There is only one commit necessary for `ledger_device_sdk` (https://github.com/alamgu/ledger-nanos-sdk/commit/c55d05853501b44c93c36599f83f71c0ed1e52cd). The patches for `ledger_secure_sdk_sys`, and `include_gif` were needed for making `cargo build` to work, but they are identical to the `ledger_secure_sdk_sys = "1.2.0"` and `include_gif = "1.0.1"` respectively.

Reply:
There are a one more commit, but that could be removed from this branch if required.
Here are all the diffs -> https://github.com/LedgerHQ/ledger-device-rust-sdk/compare/ledger_device_sdk%401.5.1...alamgu:ledger-nanos-sdk:alamgu-branch?expand=1